src/twisted/application/twist/_twist.py

```diff
@@ -99,6 +100,12 @@ def run(twistOptions):
             fileLogObserverFactory=twistOptions["fileLogObserverFactory"],
         )
         runner.run()
+        reactor = twistOptions["reactor"]
+        try:
+            if reactor._exitSignal:
+                _exitWithSignal(reactor._exitSignal)
+        except:
+            pass
```

Comment:
In general, I agree, and I really didn't like it when I added it, though I did have a purpose in mind.  I'll explain why, and if you feel it isn't necessary I'm more than happy to remove it.

The purpose here was to preserve the existing exit semantics of twist in the event _exitWithSignal throws an exception.  While that shouldn't happen (technically, I'm not confident that os.kill wont ever throw on other platforms, I'm primarily familiar with POSIX and Linux in particular), my inability to fully verify it can't happen led me to add the catch-all.

Specifically, if an exception gets thrown, I assume it'll go up to the interpreter, and from there may or may not affect the exit status of the program.  Rather than have this new code trigger that, I caught the exception and let the program exit as it does today.